File: aula 2/personagem.java
public class Personagem {
    
    String nome;
    int vida;
    int forca;
    
    public Personagem(String nome,int vida,int forca){
        this.nome = nome;
        this.vida = vida;
        this.forca = forca;
    }
    
    void receberDano(int dano) {
        vida = vida - dano;
        System.out.println(nome + " sofreu " + dano + " de dano");
    }

    boolean estarvivo(){
        if(vida > 0){
            return true;
        } else {
            return false;
        }
    
    }
    
    String ficha() {
    return nome + "vida: " + vida + "força: " + forca +;
    }
    
    void atacar(Personagem alvo) {
    System.out.println(nome + " ataca " + alvo.nome);
    alvo.receberDano(forca);
    }
    
}

public class Main {
    public static void main(String[] args) {
        
        Personagem heroi = new Personagem("Mulan", 200, 40);
        Personagem chefe = new Personagem("Thanos", 200, 5);
        
        while (heroi.estaVivo() && chefe.estaVivo()) {
            heroi.atacar(chefe);
            System.out.println(heroi.nome + " atacou " + chefe.nome + ". Vida do chefe: " + chefe.vida);

            if (!chefe.estaVivo()) {
                System.out.println(chefe.nome + " foi derrotado! " + heroi.nome + " venceu!");
                break;
            }

            chefe.atacar(heroi);
            System.out.println(chefe.nome + " atacou " + heroi.nome + ". Vida do herói: " + heroi.vida);

            if (!heroi.estaVivo()) {
                System.out.println(heroi.nome + " foi derrotado! " + chefe.nome + " venceu!");
                break;
            }
        }
    }
}

File: aula 2/personagem.py
class personagem():
    def __init__(self, nome, vida, forca):
        self.nome = nome
        self.vida = vida
        self.forca = forca

    def receberDano(self, dano):
        self.vida = self.vida - dano
        print("%s recebeu %d de dano" %(self.nome, dano))
        if self.vida<0:
            self.vida=0
    
    def estaVivo(self):
        if self.vida>0:
            return True
        else:
            return False

    def ficha(self):
        print("\nNome: ", self.nome)
        print("Vida: ", self.vida)
        print("Força: ", self.forca)
    
    def atacar(self, alvo):
        print(self.nome, " ataca ", alvo.nome)
        alvo.receberDano(self.forca)

if __name__ == "__main__":

    jogador = personagem("Angela", 100, 10)
    chefe = personagem("Jane", 500, 40)

    print("-----Jogadores-----")
    jogador.ficha()
    chefe.ficha()
    print("-----Hora da batalha-----")
    while jogador.estaVivo() and chefe.estaVivo():
        jogador.atacar(chefe)
        if not chefe.estaVivo():
            print("\n", chefe.nome, " foi derrotada")
            break
        chefe.atacar(jogador)
        if not jogador.estaVivo():
            print("\n", jogador.nome, "foi derrotada")
            break
            

    print("-----Jogadores-----")
    jogador.ficha()
    chefe.ficha()
